src/runtime_src/core/edge/skd/sk_daemon.cpp

```diff
@@ -13,120 +13,125 @@
-      char proc_name[PNAME_LEN] = {};
-      int ret;
-      skd_inst = new xrt::skd(handle,cmd->meta_bohdl,cmd->bohdl,cmd->krnl_name,i,cmd->uuid);
+    if (pid < 0) {
+      auto procMsg = boost::format("Unable to create soft kernel process %d") % i;
+      xrt_core::message::send(severity_level::error, "SKD", procMsg.str());
+      exit(EXIT_FAILURE);
+    }
 
-      /* Install Signal Handler for the Child Processes/Soft-Kernels */
-      struct sigaction act;
-      act.sa_handler = sigLog;
-      sigemptyset(&act.sa_mask);
-      act.sa_flags = 0;
-      sigaction(SIGHUP, &act, 0);
-      sigaction(SIGINT, &act, 0);
-      sigaction(SIGQUIT , &act, 0);
-      sigaction(SIGILL, &act, 0);
-      sigaction(SIGTRAP, &act, 0);
-      sigaction(SIGABRT, &act, 0);
-      sigaction(SIGBUS, &act, 0);
-      sigaction(SIGFPE, &act, 0);
-      sigaction(SIGKILL, &act, 0);
-      sigaction(SIGUSR1, &act, 0);
-      sigaction(SIGSEGV, &act, 0);
-      sigaction(SIGUSR2, &act, 0);
-      sigaction(SIGPIPE, &act, 0);
-      sigaction(SIGALRM, &act, 0);
-      sigaction(SIGTERM, &act, 0);
+    skd_inst = std::make_unique<xrt::skd>(handle, cmd->meta_bohdl, cmd->bohdl, std::string(cmd->krnl_name), i, cmd->uuid, parent_mem_bo, mem_start_paddr, mem_size);
 
-      (void)snprintf(proc_name, PNAME_LEN, "%s%d", cmd->krnl_name, i);
-      if (prctl(PR_SET_NAME, (char *)proc_name) != 0) {
-          syslog(LOG_ERR, "Unable to set process name to %s due to %s\n", proc_name, strerror(errno));
-      }
+    // Install Signal Handler for the Child Processes/Soft-Kernels
+    struct sigaction act = {};
+    act.sa_handler = sigLog;
+    sigemptyset(&act.sa_mask);
+    act.sa_flags = 0;
+    sigaction(SIGHUP, &act, nullptr);
+    sigaction(SIGINT, &act, nullptr);
+    sigaction(SIGQUIT , &act, nullptr);
+    sigaction(SIGILL, &act, nullptr);
+    sigaction(SIGTRAP, &act, nullptr);
+    sigaction(SIGABRT, &act, nullptr);
+    sigaction(SIGBUS, &act, nullptr);
+    sigaction(SIGFPE, &act, nullptr);
+    sigaction(SIGKILL, &act, nullptr);
+    sigaction(SIGUSR1, &act, nullptr);
+    sigaction(SIGSEGV, &act, nullptr);
+    sigaction(SIGUSR2, &act, nullptr);
+    sigaction(SIGPIPE, &act, nullptr);
+    sigaction(SIGALRM, &act, nullptr);
+    sigaction(SIGTERM, &act, nullptr);
 
-      /* Start the soft kernel loop for each CU. */
-      ret = skd_inst->init();
-      if(ret) {
-	syslog(LOG_ERR, "Soft kernel initialization failed!\n");
-	goto err;
-      }
-      skd_inst->report_ready();
-      skd_inst->run();
-err:
-      syslog(LOG_INFO, "Kernel %s was terminated\n", cmd->krnl_name);
-      if(skd_inst)
-	delete skd_inst;
-      exit(EXIT_SUCCESS);
+    const auto proc_name = std::string(cmd->krnl_name) + std::to_string(i);
+    if (prctl(PR_SET_NAME, proc_name.c_str()) != 0) {
+      const auto errMsg = boost::format("Unable to set process name to %s due to %s") % proc_name % strerror(errno);
+      xrt_core::message::send(severity_level::error, "SKD", errMsg.str());
```

Comment:
Is this an error? Do we need to propagate the error code up then?

Reply:
If the process cannot be set, then we will just exit.